docs/api/ref/GetAgent.md

```diff
@@ -0,0 +1,253 @@
+
+:::
+
+Finally, if there is still no Desktop Agent available, or an issue prevent connection to it, the `getAgent()` function will reject its promise with a message from the `AgentError` enumeration.
+
+```ts
+/** 
+ * Contains constants representing the errors that can be encountered when  
+ * trying to connect to a web-based Desktop Agent with the getAgent function. 
+ */ 
+enum AgentError { 
+    /** Returned if no Desktop Agent was found by any means available or 
+     * if the Agent previously connected to is not contactable on a  
+     * subsequent connection attempt.*/
+    AgentNotFound = "AgentNotFound",
+
+    /** Returned if validation of the app identity by the Desktop Agent 
+     * Failed or the app is not being allowed to connect to the Desktop Agent 
+     * for another reason. */ 
+    AccessDenied = "AccessDenied",
+
+    /** Returned if an error or exception occurs while trying to set  
+     * up communication with a Desktop Agent. */ 
+    ErrorOnConnect = "ErrorOnConnect",
+
+    /** Returned if either the failover function itself, or what it returned,  
+     * was not the right type. */ 
+    InvalidFailover = "InvalidFailover" 
+} 
+```
+
+## Failover function
+
+Desktop Agent retrieval can time out, for instance if the DA doesn't exist or is unresponsive. The default timeout of 750 milliseconds can be overridden by setting the `timeout` field. An application may also provide a failover function which will be called if an interface cannot be retrieved or times out.
+
+Example: Decreasing the timeout and providing a failover function
+
+```js
+    const fdc3 = await getAgent({
+        appId: “myApp@yourorg.org”,
+        timeout: 250,
+        failover: async (params: GetAgentParams) => {
+            // return WindowProxy | DesktopAgent
+        }
+    }); 
+```
+
+The failover function allows an application to provide a backup mechanism for connecting to a DA. It is called only when establishment through normal procedures fails or times out.
+
+:::note
+
+If you wish to _completely override FDC3s standard mechanisms_, then do not use a failover function. Instead, simply skip the `getAgent()` call and provide your own DesktopAgent object.
+
+:::
+
+Failover functions MUST be asynchronous MUST resolve to one of the following types:
+
+1. [`DesktopAgent`](./DesktopAgent)
+    The application may choose to directly import or load code that provides a `DesktopAgent` implementation. `getAgent()` will then resolve to the provided `DesktopAgent`.
+2. [`WindowProxy`](https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-windowproxy-exotic-object) ([MDN](https://developer.mozilla.org/en-US/docs/Glossary/WindowProxy)) 
```

Comment:
I still haven't implemented the windowProxy test yet.  I'm not sure I get it.   Let's say the failover function is called, and it opens an iframe... why would it return a window proxy?  Surely the iframe then should just send WCP3Handshake back to the app window?  In which case, what is the point of returning the window proxy?